components/Subscribe/index.tsx

```diff
@@ -1,36 +1,73 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
-import { FC, useState } from "react";
+import { FC, useState, useEffect, useContext } from "react";
 import axios from "axios";
+import { Modal } from "../Modal";
+import UserInfoContext from "@/context/UserInfoContext";
+import { validateEmail } from "@/lib/helpers/validateEmail";
+import { SubscribeButton } from "./subscribeButton";
 
-export const Subscribe: FC= () => {
-	const [email, setEmail] = useState<any>("");
+export const Subscribe: FC = () => {
+	const [email, setEmail] = useState<string>("");
 	const [state, setState] = useState<any>("idle");
-	// const [errorMsg, setErrorMsg] = useState<any>(null);
+	const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+	const { userInfo } = useContext(UserInfoContext);
+	const isAuth = userInfo?.email;
+	const [errorMsg, setErrorMsg] = useState<string>("");
+
+
+	type SubscribeModal = boolean | null;
+
+	useEffect(() => {
+		if (!localStorage.getItem("subscribeOpen") || localStorage.getItem("subscribeOpen")) {
+			localStorage.setItem("subscribeOpen", JSON.stringify(true)); 
+			setIsModalOpen(true); 
+		}
+	}, []);
+
+	useEffect (() => {
+		const subscribeModal: SubscribeModal = JSON.parse(localStorage.getItem("subscribeOpen") as string); 
+		if(!subscribeModal && !isAuth) {
+			setIsModalOpen(true); 
+		}
+	}, [isAuth]);
+
+	const closeSubscribe = () => {
+		localStorage.setItem("subscribeOpen", JSON.stringify(false)); 
+		setIsModalOpen(false); 
+	}
 
 	const handleSubscribe = async (e: any) => {
 		e.preventDefault();
+
+		if (!validateEmail(email)){
+			setErrorMsg("Please enter a valid email address.")
+		} else {
+			setErrorMsg(""); 
+		}
+
 		setState("Loading");
 		try {
 			await axios.post("/api/subscribe", { email });
 			setState("Success");
+			setEmail("");
 			setTimeout(() => {
 				setState("idle");
-			}, 900);
-
-			setEmail("");
+				closeSubscribe(); 
+			}, 4000);
 		} catch (e: any) {
```

Comment:
maybe also log the event error here to trace where the error mightve happend?

Reply:
console.error("Error:", e) added this line, if that is what you meant?